src/internal/m365/onedrive/restore_caches.go

```diff
@@ -0,0 +1,116 @@
+package onedrive
```

Comment:
This file is purely lift-and-shift. it was moved from `restore.go` since it has grown from a small container of maps to a struct with constructors, populators, and other methods.  No changes to this code have been made in this PR outside of the movement.